### PR TITLE
[android] Fix compatibility issues reported by lint

### PR DESCRIPTION
**Pull request by rtsisyk**
Signed-off-by: Roman Tsisyk <roman@tsisyk.com>

### CODE REVIEW

**Review by arnaudvergnet — APPROVED** [review]

File: android/src/com/mapswithme/maps/settings/DrivingOptionsFragment.java [hunk 1]
```diff
@@ -81,25 +81,25 @@ public boolean onBackPressed()
...
 
-    Switch ferriesBtn = root.findViewById(R.id.avoid_ferries_btn);
+    SwitchCompat ferriesBtn = root.findViewById(R.id.avoid_ferries_btn);
     ferriesBtn.setChecked(RoutingOptions.hasOption(RoadType.Ferry));
     CompoundButton.OnCheckedChangeListener ferryBtnListener =
         new ToggleRoutingOptionListener(RoadType.Ferry);
     ferriesBtn.setOnCheckedChangeListener(ferryBtnListener);
 
-    Switch dirtyRoadsBtn = root.findViewById(R.id.avoid_dirty_roads_btn);
+    SwitchCompat dirtyRoadsBtn = root.findViewById(R.id.avoid_dirty_roads_btn);
```
Comment (biodranik) [inline android/src/com/mapswithme/maps/settings/DrivingOptionsFragment.java:102]: А зачем нам нужен Compat? Разве у нас используется какая-то более новая, чем 21, апишка для Switch?

Comment (rtsisyk) [reply android/src/com/mapswithme/maps/settings/DrivingOptionsFragment.java:102]: Lint и Android Studio активно предлагали поменять. Место малозначительное, решил не спорить с ними.

File: android/src/com/mapswithme/maps/maplayer/traffic/widget/TrafficButton.java [hunk 2]
```diff
@@ -36,7 +39,7 @@ private static AnimationDrawable getLoadingAnim(@NonNull ImageButton trafficBtn)
     Context context = trafficBtn.getContext();
     Resources res = context.getResources();
     final int animResId = ThemeUtils.getResource(context, R.attr.trafficLoadingAnimation);
-    return (AnimationDrawable) res.getDrawable(animResId, context.getTheme());
+    return (AnimationDrawable) Objects.requireNonNull(ResourcesCompat.getDrawable(res, animResId, context.getTheme()));
```
Comment (vng) [inline android/src/com/mapswithme/maps/maplayer/traffic/widget/TrafficButton.java:42]: nit: Do we really should write this ugly expression Objects.requireNonNull(...) even if the function is already marked as NonNull ?

File: android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java [hunk 3]
```diff
@@ -25,7 +26,7 @@ public class DividerItemDecorationWithPadding extends RecyclerView.ItemDecoratio
 
   public DividerItemDecorationWithPadding(@NonNull Context context)
   {
-    mDivider = Objects.requireNonNull(context.getDrawable(R.drawable.divider_base));
+    mDivider = Objects.requireNonNull(AppCompatResources.getDrawable(context, R.drawable.divider_base));
```
Comment (vng) [inline android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java:29]: Blood from my eyes :) assert(mDivider != null); is much better IMHO than Objects.requireNonNull(...)
Lint requires it?

**Review by vng — APPROVED** [review]

File: android/src/com/mapswithme/maps/maplayer/traffic/widget/TrafficButton.java (hunk 2 above)
Comment (rtsisyk) [reply android/src/com/mapswithme/maps/maplayer/traffic/widget/TrafficButton.java:42]: ResourcesCompat.getDrawable() is declared `@Nullable`.

File: android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java (hunk 3 above)
Comment (rtsisyk) [reply android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java:29]: `AppCompatResources.getDrawable()` is nullable. I see that `assert x` are not popular in Android code base and we throw AssertionErrors manually.

Comment (vng) [reply android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java:29]: OMG, no, do not bloat the code, keep requireNonNull if there is no other options.

My opinion, if assert(mDivider != null) suppresses lint error, better to use it. If not - ok, use requireNonNull.

Where do we throw AssertionErrors manually? What is the difference with compact single assert expression? Release will crash anyway in both cases.

Comment (rtsisyk) [reply android/src/com/mapswithme/maps/widget/recycler/DividerItemDecorationWithPadding.java:29]: > Where do we throw AssertionErrors manually? 

https://github.com/organicmaps/organicmaps/search?q=AssertionError&type=code

> What is the difference with compact single assert expression? Release will crash anyway in both cases.

AFAIK, `assert` statement is disabled by default on Android until you set  `adb shell setprop debug.assert`. Explicit `throw new AssertionError` will work.

**Review by vng — APPROVED** [review]

**Pull request merged**